scripts/news/utils.py

```diff
@@ -0,0 +1,196 @@
+
+    def __init__(self, *args, **kwargs):
+        self.not_required_if = kwargs.pop("not_required_if")
+        assert self.not_required_if, "'not_required_if' parameter required"  # noqa: S101
+        kwargs["help"] = (
+            kwargs.get("help", "")
+            + " NOTE: This argument is mutually exclusive with %s" % self.not_required_if
+        ).strip()
+        super(NotRequiredIf, self).__init__(*args, **kwargs)
+
+    def handle_parse_result(
+        self, ctx: Context, opts: Mapping[str, Any], args: List[str]
+    ) -> Tuple[Any, List[str]]:
+        """Check if option is mutually exclusive with another, if yes print error and exist."""
+        other_present = self.not_required_if in opts
+
+        if other_present:
+            we_are_present = self.name in opts
+            if we_are_present:
+                err(
+                    f"{ERROR_MSG_PREFIX} Illegal usage. `%s` is mutually exclusive with `%s`"
+                    % (self.name, self.not_required_if),
+                    fg="red",
+                )
+                ctx.exit(code=1)
+            else:
+                self.prompt = None
+
+        return super(NotRequiredIf, self).handle_parse_result(ctx, opts, args)
+
+
+def sanitize_section(section: str) -> str:
+    """Cleans up a section string, making it viable as a directory name."""
+    return section.replace("/", "-").lower()
+
+
+def glob_fragments(version: str, sections: List[str]) -> List[str]:
+    """Glob all news fragments present on the repo."""
+    filenames = []
+    base = os.path.join("news", version)
+
+    if version != "next":
+        wildcard = base + ".md"
+        filenames.extend(glob.glob(wildcard))
+    else:
+        for section in sections:
+            wildcard = os.path.join(base, sanitize_section(section), "*.md")
+            entries = glob.glob(wildcard)
+            entries.sort(reverse=True)
+            deletables = [x for x in entries if x.endswith("/README.md")]
+            for filename in deletables:
+                entries.remove(filename)
+            filenames.extend(entries)
+
+    return filenames
+
+
+def get_metadata_from_file(path: Path) -> dict:
+    """Get metadata information from a news entry."""
+    new_fragment_file = path.stem
+    date, gh_pr, news_type, nonce = new_fragment_file.split(".")
+
+    with open(path, "r", encoding="utf-8") as file:
+        news_entry = file.read()
+
+    return {
+        "date": date,
+        "gh_pr": gh_pr,
+        "news_type": news_type,
+        "nonce": nonce,
+        "news_entry": news_entry,
+    }
+
+
+def get_project_meta() -> Tuple[str, str]:
+    """Get the project version and name from pyproject.toml file."""
+    with open("pyproject.toml", "rb") as pyproject:
+        file_contents = tomli.load(pyproject)
+
+    version = file_contents["tool"]["poetry"]["version"]
+    name = file_contents["tool"]["poetry"]["name"]
+    return name, version
+
+
+def load_toml_config() -> Dict[str, Any]:
```

Comment:
Perhaps this can return a custom dictionary which provides a helpful user-friendly error when a value is missing from the configuration.

Reply:
We are only keeping the changelog types in the config so I don't think that's needed and we know the structure of the config and this isn't gonna be public anyways.